src/unix/linux/disk.rs

```diff
@@ -83,35 +85,36 @@ impl DiskInner {
         self.is_read_only
     }
 
-    pub(crate) fn refresh(&mut self) -> bool {
-        self.efficient_refresh(&disk_stats())
+    pub(crate) fn refresh_specifics(&mut self, refresh_kind: DiskRefreshKind) -> bool {
+        self.efficient_refresh(refresh_kind, &disk_stats(&refresh_kind))
     }
 
-    fn efficient_refresh(&mut self, procfs_disk_stats: &HashMap<String, DiskStat>) -> bool {
-        let Some((read_bytes, written_bytes)) = procfs_disk_stats
-            .get(&self.actual_device_name)
-            .map(|stat| (stat.sectors_read * SECTOR_SIZE, stat.sectors_written * SECTOR_SIZE))
-        else {
-            sysinfo_debug!("Failed to update disk i/o stats");
-            return false;
+    fn efficient_refresh(&mut self, refresh_kind: DiskRefreshKind, procfs_disk_stats: &HashMap<String, DiskStat>) -> bool {
+        if refresh_kind.usage() {
+            let Some((read_bytes, written_bytes)) = procfs_disk_stats
+                .get(&self.actual_device_name)
+                .map(|stat| (stat.sectors_read * SECTOR_SIZE, stat.sectors_written * SECTOR_SIZE))
+                else {
+                    sysinfo_debug!("Failed to update disk i/o stats");
+                    return false;
```

Comment:
Just realized but it should not return here.

Reply:
ack, fixed.